requirements/extra.txt

```diff
@@ -1,9 +1,9 @@
 # extended list of package dependencies to reach full functionality
 
-matplotlib>3.1
-torchtext>=0.9.*
-omegaconf>=2.0.5
-hydra-core>=1.0.5
-jsonargparse[signatures]>=4.7.1
-gcsfs>=2021.5.0
-rich>=10.2.2,!=10.15.*
+matplotlib>3.1, <=3.5.1
+torchtext>=0.9.*, <=0.12.0
+omegaconf>=2.0.5, <=2.1.*
```

Comment:
noob question: How is it different from specifying the exact version? e.g. `<=2.1.*` vs. `<=2.1.2`

Not sure why some use `.*` and others use exact versions.

Reply:
`.*` includes all bugfixes...
`<=2.1.*` is the same as `<2.2`

Reply:
Should all upper bounds be written that way then?

For example,
```diff
- matplotlib>3.1, <=3.5.1
- torchtext>=0.9.*, <=0.12.0
+ matplotlib>3.1, <=3.5.*
+ torchtext>=0.9.*, <=0.12.*
```

Reply:
I think it's nothing critical, but I'm just curious why some are written so while others aren't.

Reply:
we can convert them to a specific version :)

Reply:
IMHO, we should have specific versions for all packages to further reduce the possibility of something getting broken.